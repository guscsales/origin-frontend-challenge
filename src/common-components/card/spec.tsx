import * as React from 'react';
import { mount } from 'enzyme';
import Card from '.';
import theme from '../../configs/styles/themes';
import { ThemeProvider } from 'styled-components';
import { create } from 'react-test-renderer';
import devices from '../../configs/styles/devices';

describe('Card', () => {
  const getWrapper = (props: any = {}, renderType: any = mount) =>
    renderType(
      <ThemeProvider theme={theme}>
        <Card {...props}>{props.children}</Card>
      </ThemeProvider>
    );

  describe('#render', () => {
    it('should exists the element', () => {
      const wrapper = getWrapper({ children: 'test' });

      expect(wrapper.exists()).toBe(true);
    });

    it('should return a `Card` with footer', () => {
      const wrapper = getWrapper({ footer: 'test footer', children: 'test' });

      expect(wrapper.html().indexOf('test footer') !== -1).toBe(true);
    });

    it('should return a `Card` with mobile styles', () => {
      const wrapper = getWrapper(
        { layout: true, children: 'test' },
        create
      ).toJSON();

      expect(wrapper).toHaveStyleRule('height', '100%', {
        media: devices.mobile
      });
    });

    it('should match a snapshot of a layout form', () => {
      const wrapper = getWrapper(
        { layout: true, children: 'test' },
        create
      ).toJSON();

      expect(wrapper).toMatchSnapshot();
    });

    it('should match a snapshot of a default form', () => {
      const wrapper = getWrapper({ children: 'test' }, create).toJSON();

      expect(wrapper).toMatchSnapshot();
    });
  });
});
