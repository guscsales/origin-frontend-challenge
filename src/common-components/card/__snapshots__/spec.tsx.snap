// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card #render should match a snapshot of a default form 1`] = `
.c0 {
  background: #fff;
  border-radius: 7px;
  border: 1px solid #e1e8ed;
}

@media (min-width:1024px) {
  .c0 {
    height: auto;
    border: 1px solid #e1e8ed;
  }
}

@media (min-width:320px) {
  .c1 {
    padding: 2rem 2.5rem;
  }
}

@media (min-width:1024px) {
  .c1 {
    padding: 4rem;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    test
  </div>
</div>
`;

exports[`Card #render should match a snapshot of a layout form 1`] = `
.c0 {
  background: #fff;
  border-radius: 7px;
  border: 1px solid #e1e8ed;
}

@media (min-width:320px) {
  .c0 {
    box-shadow: 0 0 3px rgba(0,0,0,0.02);
    height: 100%;
    border-radius: 10px 10px 0 0;
    border: none;
  }
}

@media (min-width:1024px) {
  .c0 {
    height: auto;
    border: 1px solid #e1e8ed;
  }
}

@media (min-width:320px) {
  .c1 {
    padding: 2rem 2.5rem;
  }
}

@media (min-width:1024px) {
  .c1 {
    padding: 4rem;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    test
  </div>
</div>
`;
